modules/widgets/migrate-to-core/gallery-widget.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * Migration from Jetpack's Gallery Widget to WordPress' Core Gallery Widget.
+ *
+ * @since 5.5
+ *
+ * @package Jetpack
+ */
+/**
+ * Migrates all active instances of Jetpack's Gallery widget to Core's Media Gallery widget.
+ */
+function jetpack_migrate_gallery_widget() {
+	// Only trigger the migration from wp-admin and outside unit tests
+	if ( ! is_admin() || defined( 'PHPUNIT_JETPACK_TESTSUITE' ) ) {
+		return;
+	}
+
+	// Only migrate if the new widget is available and we haven't yet migrated
+	if ( ! class_exists( 'WP_Widget_Media_Gallery' ) || Jetpack_Options::get_option( 'gallery_widget_migration' ) ) {
+		return;
+	}
+
+	$old_widgets = get_option( 'widget_gallery', array() );
+	$media_gallery = get_option( 'widget_media_gallery', array() );
```

Comment:
Do we expect this to return anything but null? We might want to exit early here and alert if find anything other than an empty array.

Reply:
Isn't it possible that the user has Jetpack inactive while updating to 4.9, creates a core Media Gallery (maybe because they're aware of the new features and aware of current Jetpack Gallery widgets on the site), and only activated/connected Jetpack afterwards?